File: frontend/lib/models/syslog/syslog_severity.dart
/// Syslog severity levels as specified in the RFC 5424 https://www.rfc-editor.org/rfc/rfc5424
enum SyslogSeverity {
  emerg,
  alert,
  crit,
  err,
  warning,
  notice,
  info,
  debug;

  /// Returns the corresponding SyslogSeverity corresponding with the numeric value, or null if out of range
  static SyslogSeverity? fromInt(int value) {
    if (value >= 0 && value <= 7) {
      return SyslogSeverity.values[value];
    }

    return null;
  }

  /// Returns the corresponding SyslogSeverity corresponding with the string, or null if out of range
  static SyslogSeverity? fromString(String name) {
    try {
      return SyslogSeverity.values.byName(name);
    }
    catch (_) {
      return null;
    }
  }

  @override 
  String toString() {
    return super.toString().substring("SyslogSeverity.".length);
  }
}

File: backend/model/fact_gathering/syslog/syslog_severity.py
from enum import Enum
from typing import override


class SyslogSeverity(Enum):
    emerg   = 0
    alert   = 1
    crit    = 2
    err     = 3
    warning = 4
    notice  = 5
    info    = 6
    debug   = 7

    @staticmethod
    def values() -> list["SyslogSeverity"]:
        return [name.value for name in SyslogSeverity]


    @staticmethod
    def from_json(json: dict) -> list["SyslogSeverity"]:
        return [SyslogSeverity[severity] for severity in json["severity"]]


    def __str__(self) -> str:
        return self.name